### PR TITLE
Remove GCP cluster api provider

### PR DESCRIPTION
**Pull request by patrickdillon**
I ended up doing this work while trying to eliminate our need to `replace` the cluster-api-provider-azure. More work is needed beyond this to achieve that, but we might as well commit it. I am doing this work for Azure, but we should be able to do something similar for all other providers that have migrated to openshift/api.

### CODE REVIEW

**Comment by patrickdillon** [comment]
/test e2e-gcp

I thought these tests should be run automatically with recent changes, but it looks like they are not getting queued.

File: pkg/asset/machines/master.go [hunk 1]
```diff
@@ -537,7 +535,6 @@ func (m *Master) Machines() ([]machineapi.Machine, error) {
 	awsapi.AddToScheme(scheme)
 	azureapi.AddToScheme(scheme)
 	baremetalapi.AddToScheme(scheme)
-	gcpapi.AddToScheme(scheme)
```
Comment (kirankt) [inline pkg/asset/machines/master.go:540]: I think we will need to add the GCP types into the schema here.
e.g. https://github.com/openshift/installer/blob/c3a4dbc69860b9b66b019461ce3fc3c546e1914f/pkg/asset/machines/master.go#L545-L548

File: pkg/asset/machines/worker.go [hunk 2]
```diff
@@ -555,7 +552,6 @@ func (w *Worker) MachineSets() ([]machineapi.MachineSet, error) {
 		awsprovider.SchemeGroupVersion,
 		azureprovider.SchemeGroupVersion,
 		baremetalprovider.SchemeGroupVersion,
-		gcpprovider.SchemeGroupVersion,
```
Comment (kirankt) [inline pkg/asset/machines/worker.go:558]: Ditto.

**Review by kirankt — COMMENTED** [review]
Thanks @patrickdillon I actaully meant to work on this today as well. Left a few comments. PTAL.

File: pkg/tfvars/gcp/gcp.go [hunk 3]
```diff
@@ -4,8 +4,7 @@ import (
 	"encoding/json"
 	"fmt"
 
-	gcpprovider "github.com/openshift/cluster-api-provider-gcp/pkg/apis/gcpprovider/v1beta1"
-
+	machineapi "github.com/openshift/api/machine/v1beta1"
```
Comment (staebler) [inline pkg/tfvars/gcp/gcp.go:7]: Leave the machineapi import in a section separate from the installer import.

**Review by staebler — COMMENTED** [review]
Please separate the vendor changes in their own commit.

**Comment by patrickdillon** [comment]
/test e2e-gcp

Ok, fixed it up based on the recommendations. Let's see if we get further in the tests now.

**Comment by patrickdillon** [comment]
install suceeds, 2 failed tests that look unrelated

/test e2e-gcp

**Review by kirankt — COMMENTED** [review]
It would be nice to see 2 separate commits. One for the code changes and the other for vendoring changes. Other than that it looks like the gcp test is passing, so lgtm.

**Comment by patrickdillon** [comment]
Oops, I lost track of this but have now updated/rebased.

**Review by kirankt — COMMENTED** [review]
GCP test passed.
/lgtm

**Comment by kirankt** [comment]
Should we skip the other tests? They're not relevant to this PR...

**Comment by patrickdillon** [comment]
> Should we skip the other tests? They're not relevant to this PR...

Sure we can 
/skip

**Comment by staebler** [comment]
98e2d2a30...3e4b438b1
* Rebased and ran `go mod tidy` to fix up the go.sum file.

/lgtm

**Comment by staebler** [comment]
3e4b438b1...830d0d73f
* Actually include the go.sum changes this time.

/lgtm

**Pull request merged**